src/java.base/share/classes/java/nio/file/Files.java

```diff
@@ -1583,7 +1584,8 @@ public static boolean isSameFile(Path path, Path path2) throws IOException {
      * @throws  SecurityException
      *          In the case of the default provider, and a security manager is
      *          installed, the {@link SecurityManager#checkRead(String) checkRead}
-     *          method is invoked to check read access to both files.
+     *          method is invoked to check read access to both files unless both
+     *          paths locate the same file
```

Comment:
This isn't quite right, the SM should be invoked when the paths are not equals.  I think the starting point is the SecurityException specified by isSameFile. It should says "... is invoked to check read access to both files when the two paths are not equal". The mismatch method will then follow with the same wording.

Reply:
Changed in 1bd9436ad7567b96c1baa14489721d93d7e7a436.